### PR TITLE
AIT-5114 Expand AWS Serverless definition with function_arn

### PR DESCRIPTION
**Pull request by yahya-mouman**
### What does this PR do?
This PR adds Function ARN for serverless as an https header sent to Telemetry. This comes from the need of having it to identify the serverless functions on our service end and resolve them for more data later on.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/api/telemetry.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
 )
 
+const FunctionARNKey = "function_arn"
```
Comment (maxday) [inline pkg/trace/api/telemetry.go:24]: I think you can lowercase this const as it's only used in this package?

Comment (yahya-mouman) [reply pkg/trace/api/telemetry.go:24]: Okay will do

**Review by maxday — APPROVED** [review]
🚀

**Review by paullegranddc — APPROVED** [review]

File: pkg/trace/api/telemetry.go [hunk 2]
```diff
@@ -91,6 +93,9 @@ func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
 		}
 		req.Header.Set("DD-Agent-Hostname", r.conf.Hostname)
 		req.Header.Set("DD-Agent-Env", r.conf.DefaultEnv)
+		if r.conf.GlobalTags[functionARNKey] != "" {
+			req.Header.Set("DD-Function-ARN", r.conf.GlobalTags[functionARNKey])
+		}
```
Comment (paullegranddc) [inline pkg/trace/api/telemetry.go:98]:
```suggestion
		if arn, ok := r.conf.GlobalTags[functionARNKey]; ok {
			req.Header.Set("DD-Function-ARN", key)
		}
```

File: pkg/trace/api/telemetry.go [hunk 3]
```diff
@@ -91,6 +93,9 @@ func (r *HTTPReceiver) telemetryProxyHandler() http.Handler {
 		}
 		req.Header.Set("DD-Agent-Hostname", r.conf.Hostname)
 		req.Header.Set("DD-Agent-Env", r.conf.DefaultEnv)
+		if arn, ok := r.conf.GlobalTags[functionARNKey]; ok {
+			req.Header.Set("DD-Function-ARN", key)
+		}
```
Comment (dianashevchenko) [inline pkg/trace/api/telemetry.go:98]: This part doesn't compile as the `key` variable isn't present. I believe you meant to use `arn`?

**Review by dianashevchenko — CHANGES_REQUESTED** [review]

File: pkg/trace/api/telemetry.go (hunk 3 above)
Comment (dianashevchenko) [reply pkg/trace/api/telemetry.go:98]:
```suggestion
		if arn, ok := r.conf.GlobalTags[functionARNKey]; ok {
			req.Header.Set("DD-Function-ARN", arn)
		}
```

**Review by dianashevchenko — APPROVED** [review]

**Pull request merged**